### PR TITLE
isync: 1.4.4 -> 1.5.0

### PR DESCRIPTION
**Pull request by felipeqq2**
## Description of changes

"Buffer too small" error can now be fixed by defining `QPRINTF_BUFF` (see [isync mailing list](https://sourceforge.net/p/isync/mailman/isync-devel/thread/87fsevvebj.fsf%40steelpick.2x.cz/#msg37731590)).
Patch on unexpected EOF is part of new version.

Release notes: https://sourceforge.net/projects/isync/files/isync/1.5.0/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/isync/default.nix [hunk 1]
```diff
@@ -8,20 +8,16 @@
 
 stdenv.mkDerivation rec {
```
Comment (peterhoeg) [inline pkgs/tools/networking/isync/default.nix:9]: Do you mind migrating to the `finalAttrs` pattern while at it?

File: pkgs/tools/networking/isync/default.nix [hunk 2]
```diff
@@ -6,22 +6,17 @@
...
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "isync";
-  version = "1.4.4";
+  version = "1.5.0";
 
   src = fetchurl {
-    url = "mirror://sourceforge/isync/${pname}-${version}.tar.gz";
-    sha256 = "1zq0wwvmqsl9y71546dr0aygzn9gjjfiw19hlcq87s929y4p6ckw";
+    url = "mirror://sourceforge/isync/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
```
Comment (jian-lin) [inline pkgs/tools/networking/isync/default.nix:14]:
```suggestion
    url = "mirror://sourceforge/isync/isync-${finalAttrs.version}.tar.gz";
```

Using `finalAttrs.pname` here is a bad idea.  If `pname` is overridden like this, this url becomes invalid.

```nix
isync.overrideAttrs (old: { pname = old.pname + "-patched"; })
```

I think it is better to use `pname` with `rec` or just hardcode `isync`.

**Review by soispha — APPROVED** [review]
Looks good to me. (Minor nit: You should also add the link to the mailing list either to the commit message or the comment over `-DQPRINTF_BUFF` so it's not vendor locked to the GitHub pr)

**Comment by felipeqq2** [comment]
> Looks good to me. (Minor nit: You should also add the link to the mailing list either to the commit message or the comment over `-DQPRINTF_BUFF` so it's not vendor locked to the GitHub pr)

Thanks for your review and tip! The link was added to the commit message :)

**Review by soispha — APPROVED** [review]
LGTM

**Review by axelf4 — APPROVED** [review]

**Review by soispha — APPROVED** [review]

File: pkgs/tools/networking/isync/default.nix [hunk 3]
```diff
@@ -6,22 +6,17 @@
...
+stdenv.mkDerivation (finalAttrs: {
   pname = "isync";
-  version = "1.4.4";
+  version = "1.5.0";
 
   src = fetchurl {
-    url = "mirror://sourceforge/isync/${pname}-${version}.tar.gz";
-    sha256 = "1zq0wwvmqsl9y71546dr0aygzn9gjjfiw19hlcq87s929y4p6ckw";
+    url = "mirror://sourceforge/isync/isync-${finalAttrs.version}.tar.gz";
+    sha256 = "oMgeEJOHvyedoWFFMQM5nneUav7PXFH5QTxedzVX940=";
```
Comment (peterhoeg) [inline pkgs/tools/networking/isync/default.nix:15]: Please change this to `hash = "sha256-11111111762490922d0fe8a16ec288c7cba97abbaf8=";` (or whatever the proper SRI hash is).

File: pkgs/tools/networking/isync/default.nix [hunk 4]
```diff
@@ -6,22 +6,17 @@
...
   };
 
-  patches = [
-    # Fixes "Fatal: buffer too small" error
-    ./0001-Increase-imap_vprintf-buffer-size.patch
-    # Fix #202595: SSL error "Socket error: ... unexpected eof while reading"
-    # Source: https://sourceforge.net/p/isync/isync/ci/b6c36624f04cd388873785c0631df3f2f9ac4bf0/
-    ./work-around-unexpected-EOF-error-messages-at-end-of-SSL-connections.patch
-  ];
+  # Fixes "Fatal: buffer too small" error
```
Comment (peterhoeg) [inline pkgs/tools/networking/isync/default.nix:18]: Do we have a link to an upstream bug/issue to put here?

**Review by peterhoeg — CHANGES_REQUESTED** [review]

**Review by peterhoeg — APPROVED** [review]

**Comment by Jdogzz** [comment]
I've got bitten by a bug introduced by this new version, looks like there were a number of patches submitted after 1.5.0 was released that are getting picked up by others:

https://bugzilla.redhat.com/show_bug.cgi?id=2314719

**Comment by peterhoeg** [comment]
@Jdogzz, any chance you could do up a PR with the required patches?

**Comment by Jdogzz** [comment]
@peterhoeg I've now tried my hand at it in #345713, please let me know if there's a preferable way to do so in that thread.

**Pull request merged**